gravitee-am-service/src/main/java/io/gravitee/am/service/validators/domain/DomainValidatorImpl.java

```diff
@@ -59,6 +57,26 @@ public Completable validate(Domain domain, List<String> domainRestrictions) {
             return Completable.error(new InvalidDomainException("Domain can only work in vhost mode"));
         }
 
+        Completable error = checkVhost(domain, domainRestrictions, chain);
+        if (error != null) {
+            return error;
+        }
+
+        if (domain.getWebAuthnSettings() != null && domain.getWebAuthnSettings().getCertificates() != null) {
+            boolean containsInvalidCertFormat = domain.getWebAuthnSettings()
+                    .getCertificates()
+                    .values()
+                    .stream()
+                    .anyMatch(cert -> !(cert instanceof String certValue) || (certValue.startsWith("-----") || certValue.endsWith("-----")));
+            if (containsInvalidCertFormat) {
+                return Completable.error(new InvalidDomainException("WebAuthnSettings contains certificates with boundaries"));
+            }
+        }
+
+        return Completable.merge(chain);
+    }
+
+    private Completable checkVhost(Domain domain, List<String> domainRestrictions, List<Completable> chain) {
```

Comment:
TODO: the new version of this method is less easy to understand IMO. "checkVHost" is not just checking but also update the "chain" list. IMO checkVHost shouldn't take chain as parameter and return Completable.merge(chain), on the checkVHost response we can then do a .andTHen call to manage the webAuthn validation.

Reply:
fixed